python/ray/tune/test/cluster_tests.py

```diff
@@ -71,21 +84,18 @@ def start_connected_emptyhead_cluster():
                 "num_heartbeats_timeout": 10
             })
         })
-    register_test_trainable()
+    # Pytest doesn't play nicely with imports
+    _register_all()
     yield cluster
     # The code after the yield will run as teardown code.
     ray.shutdown()
     cluster.shutdown()
 
 
-@pytest.mark.skipif(
```

Comment:
Hangs are usually on some redis subscription, and somehow the timeout can't interrupt that